canvasapi/assignment.py

```diff
@@ -202,6 +202,55 @@ def get_submissions(self, **kwargs):
             _kwargs=combine_kwargs(**kwargs),
         )
 
+    def set_extensions(self, assignment_extensions, **kwargs):
+        """
+        Set extensions for student assignment submissions
+
+        :calls: `POST /api/v1/courses/:course_id/assignments/:assignment_id/extensions \
+        <https://canvas.instructure.com/doc/api/assignment_extensions.html#method.assignment_extensions.create>`_
+
+        :param assignment_extensions: list of dictionaries representing extensions
+        :type assignment_extensions: list
+
+        :rtype: list of :class:`canvasapi.assignment.AssignmentExtension`
+
+        Example Usage:
+
+        >>> assignment.set_extensions([
+        ...     {
+        ...         'user_id': 3,
+        ...         'extra_attempts: 2
+        ...     },
+        ...     {
+        ...         'user_id': 2,
+        ...         'extra_attempts: 2
+        ...     }
+        ... ])
+        """
+        if not isinstance(assignment_extensions, list) or not assignment_extensions:
```

Comment:
When including a check on whether or not a variable is set as well as some property (such as instance) of that variable, do the existence check first. See the [Python docs on boolean operations for details on short-circuiting](https://docs.python.org/3/library/stdtypes.html#boolean-operations-and-or-not).

Additionally, since `assignment_extensions` is passed in as a required argument, the `not assignment_extensions` is probably not even needed.